core/trino-spi/src/main/java/io/trino/spi/SplitWeight.java

```diff
@@ -36,8 +38,11 @@ private SplitWeight(long value)
         this.value = value;
     }
 
+    /**
+     * @return The internal integer representation for this weight value
```

Comment:
It doesn't necessarily need to be just "the internal representation". We could say weights are just an integer value larger than 0. In that case, `getValue()` is the right name for this method.

I haven't looked at the rest of the code yet, but I assume that all that matters for scheduling purposes is the relative weights between splits and in comparison to the total weight assigned to each tasks.

Reply:
> I assume that all that matters for scheduling purposes is the relative weights between splits and in comparison to the total weight assigned to each tasks.

In some sense, yes- the relative weight is what matters to control the effective behavior. However, this is accomplished by making the weight relative to a "standard weight" and not just the relative weight between to individual splits. Most logic that uses the long value is just adding or subtracting these values from some total sum, _except_ for the initial scheduler parameters configured via `NodeSchedulerConfig(node-scheduler.max-pending-splits-per-task)` and `NodeSchedulerConfig(node-scheduler.max-splits-per-node)`. Critically, that point needs to be able to say that a configured split count of N is equivalent to a combined weight of X "standard" splits.

> It doesn't necessarily need to be just "the internal representation".

Yeah, I'm not in love with the name, and am happy to change this back to `getValue()`, but what I'm trying to achieve is a de-coupling between SPI consumers, the scheduler logic, and some chosen integer value for a "standard split" so that we can in the future choose to redefine `UNIT_VALUE` (currently: 100) to some alternative value (eg: 1,000) without breaking SPI consumers. This was raised during review by @arhimondr as a consideration (that I think is valid), that we might in the future need to express relative weights between splits at a higher granularity than a value of 100 allows. To achieve that, there are few important interaction points to disambiguate from one another to keep the the "relative to standard" abstraction encapsulated:
- `SplitWeight#encodedValueForStandardSplitCount` for the scheduler to translate its configured split counts into the integer equivalent (again, not in love with the name and happy to change it)
- `SplitWeight#fromProportion(double)` for connectors to translate their weighting scheme into a "relative to standard" without hard coding information about the integer value of a standard split, ie: 0.5 should be 1/2 of the standard and 2.0 should be 2x the standard weight, regardless of what the standard weight is chosen to be at any point in time (happy to change the name here too)
- `SplitWeight#getEncodedValue()` where the logic is simply doing weight accounting, adding and subtracting this value on a per-split basis, and doesn't need to understand any association of an integer relative to standard to weight (ie: most of the code in `trino-main`, with the notable exception of the scheduler)
- `SplitWeight#toString()` and maybe another method in the future for a connector to recover some "relative to standard" view of a weight value for the purposes of logging / debugging without coupling to an assumption of `UNIT_VALUE = 100`, which might be too tempting if this method were just named `getValue()`.